cocos/2d/CCSprite.cpp

```diff
@@ -372,11 +381,48 @@ void Sprite::setVertexLayout()
     vertexLayout.setLayout(sizeof(V3F_C4B_T2F), backend::VertexStepMode::VERTEX);
 }
 
+
+
 void Sprite::updateShaders(const char* vert, const char* frag)
 {
-    auto programState = new (std::nothrow) backend::ProgramState(vert, frag);
-    setProgramState(programState);
-    CC_SAFE_RELEASE_NULL(programState);
+    if(_defaulCommandInfo.ps != nullptr && vert == positionTextureColor_vert && frag == positionTextureColor_frag)
+    {
+        auto programState = _defaulCommandInfo.ps->clone();
```

Comment:
`ProgramState::clone` is faster than `ProgramState::ProgramState(const char*, const char *)`

Reply:
This is strange....
There is already in ProgramState::ProgramState(const char*, const char *) one cache
` _program = backend::ProgramCache::getInstance()->newProgram`
and with this PR we would like to have another cache on top of backend::ProgramCache?

Reply:
We have discussed the performance issue of ProgramCache yesterday, and we will fix that later.

Reply:
Since we will optimize ProgramCache, this PR is not needed.